internal/tools/aws/s3.go

```diff
@@ -150,6 +234,25 @@ func (a *Client) S3EnsureBucketDeleted(bucketName string, logger log.FieldLogger
 		logger.WithField("s3-bucket-name", bucketName).WithError(err).Warn("Could not determine if S3 bucket exists")
 	}
 
+	isVersioned, err := a.S3IsVersioningEnabled(bucketName)
+
+	if err != nil {
+		logger.WithField("s3-bucket-name", bucketName).WithError(err).Warn("Could not determine if S3 bucket is versioned")
+		// TODO: Continue anyways?
+		return err
+	}
```

Comment:
@gabrieljackson thoughts here? Should we just assume not versioned in this case?